strawberry/field.py

```diff
@@ -37,7 +37,7 @@
     from .object_type import TypeDefinition
 
 
-_RESOLVER_TYPE = Union[StrawberryResolver, Callable]
+_RESOLVER_TYPE = Union[StrawberryResolver, Callable, staticmethod]
```

Comment:
is staticmethod a type? :O

Reply:
Yes, `staticmethod` is a class and its objects are not `Callable`

Reply:
oh, interesting! thanks 😊

Reply:
> its objects are not `Callable`

I don't think this is true:
```pycon
>>> class Blah:
...     @staticmethod
...     def foo(): ...
... 
>>> callable(Blah.foo)
True
>>> callable(Blah().foo)
True
```

Mypy is perfectly happy with `staticmethod`s as `Callable` too:
https://mypy-play.net/?mypy=latest&python=3.10&gist=c907cc3700c23323978d2ec618a724dc

Reply:
Oh, I think I understand. The issue is during class evaluation, where the function is actually an instance of `staticmethod` and not just of `function`

Reply:
> > its objects are not `Callable`
> 
> I don't think this is true:

Looks like static methods were made callables in 3.10: https://bugs.python.org/issue43682.

I tested in this way:
```python
>>> from collections.abc import Callable
>>> isinstance(staticmethod(int), Callable)
```

This returns `True` for Python 3.10 and `False` for previous versions.